File: CsChatClient/Messages/VideoMessageContent.cs
namespace CsChatClient.Messages
{
    [ContentAttribute(MessageContentType.VoipContentTypeStart, MessageContentPersistFlag.PersistFlagPersistAndCount)]
    public class VideoMessageContent : MediaMessageContent
    {
        public byte[] ThumbnailData { get; set; }

        public override void Decode(MessagePayload payload)
        {
            base.Decode(payload);
            ThumbnailData = payload.BinaryContent;
        }

        public override string Digest(MessageEx message)
        {
            return "[视频]";
        }

        public override MessagePayload Encode()
        {
            MessagePayload payload = base.Encode();
            payload.BinaryContent = ThumbnailData;
            return payload;
        }
    }
}

File: CppChatClient/message/VideoMessageContent.cpp
//
//  VideoMessageContent.cpp
//  ChatClient
//
//  Created by Hao Jia on 2020/1/31.
//  Copyright © 2020 WF Chat. All rights reserved.
//

#include "VideoMessageContent.h"

namespace WFCLib {

VideoMessageContentPrototype VideoMessageContent::sPrototype;

MessagePayload VideoMessageContent::encode() const {
    MessagePayload payload = MediaMessageContent::encode();
    payload.binaryContent = thumbnail;
    return payload;
}

void VideoMessageContent::decode(const MessagePayload & payload) {
    MediaMessageContent::decode(payload);
    thumbnail = payload.binaryContent;
}

MessageContent* VideoMessageContent::clone() const {
    VideoMessageContent *p = new VideoMessageContent();
    p->extra = extra;
    p->thumbnail = thumbnail;
    p->localPath = localPath;
    p->remoteUrl = remoteUrl;
    return p;
}

const MessageContentPrototype* VideoMessageContent::getPrototype() const {
    return &sPrototype;
}

std::string VideoMessageContent::digest() const {
    return "[视频]";
}
};
